src/proposals/proposals.controller.ts

```diff
@@ -215,49 +344,86 @@ export class ProposalsController {
     );
   }
 
-  // DELETE /proposals/:id/attachments/:fk
+  // DELETE /proposals/:pid/attachments/:aid
   @UseGuards(PoliciesGuard)
   @CheckPolicies((ability: AppAbility) =>
     ability.can(Action.Delete, Attachment),
   )
-  @Delete("/:id/attachments/:fk")
+  @Delete("/:pid/attachments/:aid")
+  @ApiParam({
+    name: "pid",
+    description: "Id of the proposal for which we would like to delete the attachment specified",
+    type: String,
+  })
+  @ApiParam({
+    name: "aid",
+    description: "Id of the attachment of this proposal that we would like to delete",
+    type: String,
+  })
+  @ApiResponse({
+    status: 200,
+    description: "Remove the attachment with id specified associated with the proposal with the pid specified"
+  })
   async findOneAttachmentAndRemove(
-    @Param("id") proposalId: string,
-    @Param("fk") attachmentId: string,
+    @Param("pid") proposalId: string,
+    @Param("aid") attachmentId: string,
   ): Promise<unknown> {
     return this.attachmentsService.findOneAndRemove({
       _id: attachmentId,
       proposalId: proposalId,
     });
   }
 
-  // POST /proposals/:id/datasets
-  @UseGuards(PoliciesGuard)
+  // POST /proposals/:pid/datasets
+/*   @UseGuards(PoliciesGuard)
   @CheckPolicies((ability: AppAbility) =>
     ability.can(Action.Create, DatasetClass),
   )
-  @Post("/:id/datasets")
+  @Post("/:pid/datasets")
+  @ApiParam({
+    name: "id",
+    description: "Id of the proposal for which we would like to retrieve all the datasets",
+    type: String,
+  })
+  @ApiResponse({
+    status: 200,
+    type: DatasetClass,
+    isArray: true,
+    description: "Array with all the datasets associated with the proposal with the pid specified"
+  })
   async createDataset(
-    @Param("id") id: string,
+    @Param("pid") id: string,
     createRawDatasetDto: CreateRawDatasetDto,
   ): Promise<DatasetClass> {
     const createRawDataset = { ...createRawDatasetDto, proposalId: id };
     return await this.datasetsService.create(createRawDataset);
   }
+ */
 
   // GET /proposals/:id/datasets
   @UseGuards(PoliciesGuard)
   @CheckPolicies((ability: AppAbility) =>
     ability.can(Action.Read, DatasetClass),
   )
-  @Get("/:id/datasets")
+  @Get("/:pid/datasets")
+  @ApiParam({
+    name: "pid",
+    description: "Id of the proposal for which we would like to retrieve all the datasets",
+    type: String,
+  })
+  @ApiResponse({
+    status: 200,
+    type: DatasetClass,
+    isArray: true,
+    description: "Array with all the datasets associated with the proposal with the pid specified"
+  })
   async findAllDatasets(
-    @Param("id") proposalId: string,
+    @Param("pid") proposalId: string,
   ): Promise<DatasetClass[] | null> {
     return this.datasetsService.findAll({ where: { proposalId } });
   }
 
-  // PATCH /proposals/:id/datasets/:fk
+/*   // PATCH /proposals/:id/datasets/:fk
```

Comment:
I see this part is also commented. Are we using it or it should be removed.

Reply:
Done!!!